### PR TITLE
ObjectStore

### PR DESCRIPTION
**Pull request by kaloyan-raev**
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/128)

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
@jtolds After removing the methods for extended attributes, there is not much left in the ObjectStore layer - it just prepends "object" to the path and calls the StreamStore, which also seems redundant in the current situation.

If we don't need extended attributes as streams, is there any other reason to have streams in the architecture of the Structured File Storage?

If no then I suggest to remove the StreamStore layer and move its responsibilities to the ObjectStore.

**Comment by kaloyan-raev** [comment]
@jtolds As discussed, the ObjectStore for now will just serialize the metadata for the StreamStore.

File: pkg/storage/list.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+package storage
+
+import (
+	"storj.io/storj/pkg/paths"
+)
+
+// ListItem is a single item in a listing
+type ListItem struct {
+	path paths.Path
+	meta Meta
```
Comment (jtolds) [inline pkg/storage/list.go]: probably need an `IsPrefix bool`

Comment (kaloyan-raev) [reply pkg/storage/list.go]: We will add such method to the Path type, so no need for a flag here.

**Pull request merged**